### PR TITLE
OboeTester: Create foreground service

### LINKED ISSUE #2065: Add a foreground service to OboeTester
Starting in Android U, [foreground services](https://developer.android.com/develop/background-work/services/foreground-services) are enforced for Microphone access when the app runs in the background.

We should add a foreground service to OboeTester so the audio can run in the background.

### PR DESCRIPTION
**Pull request by robertwu1**
Fixes #2065

### CODE REVIEW

**Review by atneya — APPROVED** [review]
lgtm

File: apps/OboeTester/app/src/main/java/com/mobileer/oboetester/MainActivity.java [hunk 1]
```diff
@@ -16,6 +16,8 @@
 
 package com.mobileer.oboetester;
 
+import static com.mobileer.oboetester.AudioForegroundService.ACTION_START;
```
Comment (philburk) [inline apps/OboeTester/app/src/main/java/com/mobileer/oboetester/MainActivity.java]: Are these used?

File: apps/OboeTester/app/src/main/java/com/mobileer/oboetester/AudioForegroundService.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.ServiceInfo;
+import android.os.Build;
+import android.os.IBinder;
+import android.util.Log;
+
+public class AudioForegroundService extends Service {
+    private static final String TAG = "AudioForegroundService";
```
Comment (philburk) [inline apps/OboeTester/app/src/main/java/com/mobileer/oboetester/AudioForegroundService.java:31]: I have been using OboeTester as the tag so I can enable all OboeTester logs.

**Review by philburk — CHANGES_REQUESTED** [review]
For testing and debugging, we need to retain the ability to test what happens if you do NOT start a ForegroundService. So please add another checkbox on MainActivity screen below "enable background operation" labelled "use foregroundService".

**Review by philburk — APPROVED** [review]
What notification does the user see?

**Comment by robertwu1** [comment]
Screenshot_20240712-110941~2
Screenshot_20240712-110951

**Pull request merged**